applications/RomApplication/tests/NonlinearTestFiles/MainKratosROM.py

```diff
@@ -0,0 +1,184 @@
+        ################################################################
+
+
+        # ################################################################
+        # #                      RADIATION PROPERTY                      #
+        # ################################################################
+
+        RadiationSettings = KratosMultiphysics.Parameters("""
+            {
+                "model_part_name": "ThermalModelPart.ThermalFace2D_Right_Wall",
+                "add_ambient_radiation": true,
+                "emissivity": 0.8,
+                "add_ambient_convection": true,
+                "convection_coefficient": 100.0
+            }
+            """
+            )
+
+        AmbientTemperature = 12.0
+        RadiationSettings.AddEmptyValue("ambient_temperature").SetDouble(AmbientTemperature)
+        self.processes.append(ApplyThermalFaceProcess(self.model, RadiationSettings))
+        ################################################################
+        for process in self.processes:
+            process.ExecuteBeforeSolutionLoop()
+        ################################################################
+
+    def EvaluateQuantityOfInterest(self):
+       ##############################################################################################
+       # Functions evaluating the QoI of the problem: Array of temperature at every node on the mesh #
+       #                    and nodal area                                                           #
+       ##############################################################################################
+        ArrayOfTemperatures = []
+        for node in self._GetSolver().GetComputingModelPart().Nodes:
+            ArrayOfTemperatures.append(node.GetSolutionStepValue(KratosMultiphysics.TEMPERATURE))
+        return ArrayOfTemperatures
+
+    def EvaluateQuantityOfInterest2(self):
+        ArrayOfAreas = []
+        for node in self._GetSolver().GetComputingModelPart().Nodes:
+            ArrayOfAreas.append(node.GetSolutionStepValue(KratosMultiphysics.NODAL_AREA))
+        return ArrayOfAreas
+
+    def FinalizeSolutionStep(self):
+        super(ConvectionDiffusionAnalysisWithFlush,self).FinalizeSolutionStep()
+        if self.parallel_type == "OpenMP":
+            now = time.time()
+            if now - self.last_flush > self.flush_frequency:
+                sys.stdout.flush()
+                self.last_flush = now
+
+        ##############################################################################################
+
+if __name__ == "__main__":
+
+    with open("ProjectParametersROM.json",'r') as parameter_file:
+        parameters = KratosMultiphysics.Parameters(parameter_file.read())
+
+    model = KratosMultiphysics.Model()
+    simulation = ConvectionDiffusionAnalysisWithFlush(model,parameters)
+    simulation.Run()
+
+    QoI2 = simulation.EvaluateQuantityOfInterest2()
+    print(QoI2)

```

Comment:
If these are debug prints, please remove them. If not, please use the Kratos logger.